langchain4j/src/main/java/dev/langchain4j/service/DefaultAiServices.java

```diff
@@ -166,11 +169,34 @@ public Object invoke(Object proxy, Method method, Object[] args) throws Exceptio
                             return new AiServiceTokenStream(messages, contents, context, memoryId); // TODO moderation
```

Comment:
streaming case is left without toolProvideer support, is it intentional?

Reply:
Oh, no. This was not intended.
I will look into this.

Reply:
What do you think about my solution, overriding the context toolSpecifications and toolExecutors?
[DefaultAiServices.java](https://github.com/langchain4j/langchain4j/pull/1644/commits/e7c2839b8a1bd113125531a2445b8aa0873f23fb#diff-1d724bf189ebfdb3d76eaefc0f9f1768b01e615ae3521ade0ea4fe4c8d774bd2R189)

Reply:
@MiggiV2 although simple, it does not seem right to override global tools with local (for this request only) ones.
What do you think about modifying `AiServiceTokenStream` and `AiServiceStreamingResponseHandler` to accept a list of tool specifications instead?

Reply:
Sounds like a more and clean solution. I was also experimenting with this, but didn't want to change so much on the code base.
I will modify both classes and will think of backwards compatibility.

Reply:
@MiggiV2 ok, thanks! Do not worry about backwards compatibility, those 2 classes are internal, noone should use them directly. Adding an extra list to both constructors should be ok